pkg/urbit/vere/io/http.c

```diff
@@ -1732,6 +1915,88 @@ _http_ef_http_server(u3_httd* htd_u,
   u3z(dat);
 }
 
+/* _http_stream_slog(): emit slog to open connections
+*/
+static void
+_http_stream_slog(void* vop_p, c3_w pri_w, u3_noun tan)
+{
+  u3_httd* htd_u = (u3_httd*)vop_p;
+  u3_hreq* seq_u = htd_u->fig_u.seq_u;
+
+  //  only do the work if there are open slog streams
+  //
+  if ( 0 != seq_u ) {
+    u3_weak data = u3_none;
+
+    if ( c3y == u3a_is_atom(tan) ) {
+      data = u3nt(u3_nul, u3r_met(3, tan), u3k(tan));
+    }
+    else {
+      u3_weak wol = u3_none;
+
+      //  if we have no arvo kernel and can't evaluate nock,
+      //  only send %leaf tanks
+      //
+      if ( 0 == u3A->roc ) {
+        if ( c3__leaf == u3h(tan) ) {
+          wol = u3nc(u3k(u3t(tan)), u3_nul);
+        }
+      }
+      else {
+        u3_noun blu = u3_term_get_blew(0);
+        c3_l  col_l = u3h(blu);
+        wol = u3dc("wash", u3nc(0, col_l), u3k(tan));
+        u3z(blu);
+      }
+
+      if ( u3_none != wol ) {
+        u3_noun low = wol;
+        u3_atom txt = u3_nul;
+        while ( u3_nul != low ) {
+          u3_atom lin = u3qc_cat(3, u3qc_rap(3, u3k(u3h(low))), c3_s2('\n', '\n'));
+          txt = u3qc_cat(3, txt, u3qc_cat(3, u3i_string("data:"), lin));
```

Comment:
These refcounts don't look quite right -- callers to `q` jet functions retain their references. I also suspect that this loop could be refactored to simply cons/weld a list of atoms and only concatenate (rap) at the end. I'd have to work through the details though.

Another possibility would be send each line separately via `_http_continue_respond()`.